puppet-mode.el

```diff
@@ -275,12 +275,19 @@ Return nil, if there is no special context at POS, or one of
   "Launch a Puppet REPL using `puppet-repl-command' as an inferior mode."
   (interactive)
 
+  ;; the command (at least for now) is 'puppet debugger'
+  ;; but really it is 'puppet' with args 'debugger' + puppet-repl-args
+  ;; Even if it changes, we need to not break on multi-word commands
   (unless (comint-check-proc puppet-repl-buffer)
-    (set-buffer
-     (apply 'make-comint "Puppet-REPL"
-            puppet-repl-command
-            nil
-            puppet-repl-args))
+    (let* ((command-list (split-string puppet-repl-command))
+           (real-puppet-repl-command (car command-list))
+           (real-puppet-repl-args (append (cdr command-list)
```

Comment:
yeah, but i can't think of any other way.

We need to normalize "puppet debugger" into puppet as the command with debugger + rest of the arguments as arguments. Or we can not allow customizing the puppet-debugger command, so it can be simplified to 

```
(apply 'make-comint "Puppet-REPL"
              "puppet"
              nil
              (cons "debugger"  puppet-repl-args)))
```
or we can take up logicminds's offer, get a single word puppet-debugger command and accept a few versions of puppet debugger will never work with this mode